src/gui/torrentcreatordialog.cpp

```diff
@@ -233,8 +229,14 @@ void TorrentCreatorDialog::onCreateButtonClicked()
         , m_ui->URLSeedsList->toPlainText().split(u'\n', Qt::SkipEmptyParts)
     };
 
-    // run the creator thread
-    m_creatorThread->create(params);
+    auto *torrentCreator = new BitTorrent::TorrentCreator(params);
+    connect(this, &QDialog::finished, torrentCreator, &BitTorrent::TorrentCreator::requestInterruption);
+    connect(torrentCreator, &BitTorrent::TorrentCreator::creationSuccess, this, &TorrentCreatorDialog::handleCreationSuccess);
+    connect(torrentCreator, &BitTorrent::TorrentCreator::creationFailure, this, &TorrentCreatorDialog::handleCreationFailure);
+    connect(torrentCreator, &BitTorrent::TorrentCreator::updateProgress, this, &TorrentCreatorDialog::updateProgressBar);
+
+    // run the torrentCreator in a thread
+    QThreadPool::globalInstance()->start(torrentCreator);
```

Comment:
IMO it should use its own `QThreadPool` instead of the global one. Who knows what tasks are scheduled there?

Reply:
Added a threadPool to the dialog class.

Reply:
In this particular case it's only about handling the (unique) torrent creation dialog in the GUI interface. I agree with @Chocobo1 that it's better to have a dedicated thread for it which doesn't consume the default QT thread pool instance. Especially considering that this may be a very long-running operation. When creating torrents via the API, in another PR, it will also be better to have a separate threadPool with a limited size. Again, to not monopolize the internal default pool for low-priority, but long-running-time jobs of creating torrents.